experimental/python/databricks/bundles/build.py

```diff
@@ -184,8 +180,53 @@ def python_mutator(
             ),
         )
 
-    conf_dict = experimental.get("python", {})
+    experimental_conf_dict = experimental.get("python", {})
+    experimental_conf = _transform(_Conf, experimental_conf_dict)
+
+    conf_dict = input.get("python", {})
     conf = _transform(_Conf, conf_dict)
+
+    has_conf = conf != _Conf()
+    has_experimental_conf = experimental_conf != _Conf()
+
+    if has_conf and not has_experimental_conf:
+        return conf, Diagnostics()
+    elif not has_conf and has_experimental_conf:
+        # do not generate warning in Python code, if CLI supports non-experimental 'python',
+        # it should generate a warning
+        return experimental_conf, Diagnostics()
+    elif has_conf and has_experimental_conf:
+        # for backward-compatibility, CLI can copy contents of 'python' into 'experimental/python'
+        # if configs are equal, it isn't a problem
+        if conf != experimental_conf:
+            return (
+                _Conf(),
+                Diagnostics.create_error(
+                    "Both 'python' and 'experimental/python' sections are present, use 'python' section only",
```

Comment:
Why not have validation in CLI itself and keep Python part light? E.g. use 'python', fallback to 'experimental.python'.

Reply:
> There is CLI validation in https://github.com/databricks/cli/pull/3540, but we can't assume that both have the latest version.

Why not enforce minimum or exact databricks-bundles version in CLI?

Reply:
For discussion: do you have ideas on how to enforce bidirectional compat for future changes? If so, do we use some kind of version range to do this? Some changes will require a coupled bump. It may be easier to just require version coupling, or at least warn if you're using different versions.